e2e/nomostest/config_sync.go

```diff
@@ -942,23 +957,37 @@ func (nt *NT) RootSyncObjectHelm(name string, chart *registryproviders.HelmPacka
 	default:
 		nt.T.Fatalf("Unrecognized HelmProvider: %s", *e2e.OCIProvider)
 	}
+	// Set the default ReconcileTimeout
+	if nt.DefaultReconcileTimeout != nil {
+		rs.Spec.SafeOverride().ReconcileTimeout = toMetav1Duration(*nt.DefaultReconcileTimeout)
+	} else if rs.Spec.Override != nil {
+		rs.Spec.Override.ReconcileTimeout = nil
+	}
+	// Enable automatic deletion of managed objects by default.
+	// This helps ensure that test artifacts are cleaned up.
+	EnableDeletionPropagation(rs)
+	// Register the Helm chart for syncing
+	rsKey := client.ObjectKeyFromObject(rs)
+	nt.RootSyncHelmCharts[rsKey] = chartID
+	// Unregister the Git repo for syncing, if any
+	delete(nt.RootRepos, rsKey.Name) // TODO: delete repo?
```

Comment:
Same here. We should delete from `nt.RootRepos` when a helm RootSync is converted from a Git RootSync.

Reply:
My thinking was that if we make a helm rsync, we should delete it from the map of git rsyncs, and vice versa. But RootRepos/NonRootRepos is acting as multiple different things: a list of RSyncs, and a list of Repositories, a map from RSync to Repo.

The TODO here is about whether we need to delete the underlying Repository when we remove the record of the RSync that uses it. However, I think the Repository is cleaned up elsewhere. So I don't think we need to do any cleanup here.

In the future, it might be cleaner to combine these maps, and make some generic Source interface with impls for Helm, OCI, & Git. But I didn't really want to do all the work now to make that happen. This effort as just a way to make the tests less flaky and kind of got a little out of hand.

Reply:
> My thinking was that if we make a helm rsync, we should delete it from the map of git rsyncs, and vice versa.

There are two ways of making a helm rsync.
- If it is a brand new helm rsync, we don't need to delete it from the map.
- If the helm rsync is converted from a Git rsync, then we should delete from the map when the conversion happens (a.k.a. where `RootSyncObjectHelm` is invoked).

Reply:
Right, but delete is just a no-op if the key doesn't exist in the map. So we don't need to check that it exists before deleting it.

Reply:
It is okay that we only have `nt.RootSyncHelmCharts` and `nt.RootRepos` for now by deleting the other one. It may be a bit messier if we're going to have `nt.RootSyncOCIRepos` in the future.

Looks good to me now, but just a headsup when we need to refactor a gain later.

Reply:
I'll take another stab this week at some incremental improvements.